core-inline-blocks/inline-image/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { MediaUpload } from '@wordpress/editor';
+
+export const name = 'core/inline-image';
+
+export const settings = {
+	id: 'inline-image',
+
+	title: __( 'Inline Image' ),
+
+	type: 'image',
+
+	icon: 'format-image',
+
+	category: 'inline',
+
+	edit( { onSave } ) {
```

Comment:
I like how you added `edit` and `save` functions to provide an interface similar to regular blocks. I wonder if it will be possible to have the edit toolbar in https://github.com/WordPress/gutenberg/pull/5794#issuecomment-381516924 be part of this `edit` function as well? Not sure if this is possible though, since I think the React element returned from the `save` function will be rendered to string before using the edit toolbar.

Reply:
Probably will have to be some different method. We can name these differently based on what it's for. Maybe insert, edit/select, and save or similar.